git/sparse_checkout.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+URL=$1
+
+if [ -z "$1" ]; then
+  echo "Usage [REMOTE-URL]"
+  exit 1
+fi
+
+REMOTE="$(echo "$URL" | sed -n 's|\(.*github.com\)/\([^/]*\)/\([^/]*\)/\(.*\)|\1/\2/\3.git|p')"
+MAIN_DIR="$(echo "$URL" | sed -n 's|\(.*github.com\)/\([^/]*\)/\([^/]*\)/\(.*\)|\3|p')"
+SUBDIR="$(echo "$URL" | sed -n 's|\(.*github.com\)/\([^/]*\)/\([^/]*\)/\(.*\)|\4|p')"
+
+test -d "$MAIN_DIR" || mkdir "$MAIN_DIR"
+cd "$MAIN_DIR" || exit
```

Comment:
Why do we need || exit?

Reply:
good practice, if cd fails instead of carrying out the rest of the operations, we just exit

Reply:
okay, but since we're creating the folder in the line above, cd would never fail?

Reply:
true, its more of a habitual thing, I can remove it if necessary